sdk/eventhub/Azure.Messaging.EventHubs/stress/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# right version of dotnet?
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env 
+
+# Copy in engineering system needed to build - TODO: cut this down to just what is needed
+COPY /sdk/eventhub/Azure.Messaging.EventHubs/ /app/sdk/eventhub/Azure.Messaging.EventHubs/
+COPY /eng /app/eng
+COPY /tools /app/tools
+COPY ./*.proj /app
+COPY ./Directory.Build.props /app
+COPY ./Directory.Build.targets /app
+COPY ./NuGet.Config /app
+COPY ./sdk/eventhub/Azure.Messaging.EventHubs.Shared /app/sdk/eventhub/Azure.Messaging.EventHubs.Shared
+COPY ./sdk/core /app/sdk/core
+
+# Run restore and build, copy the .dll files over so they can used in the running container
+WORKDIR /app
+RUN dotnet restore './sdk/eventhub/Azure.Messaging.EventHubs/stress/src/'
+RUN dotnet build './sdk/eventhub/Azure.Messaging.EventHubs/stress/src/'
+COPY /artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug /app/bin/Debug
+
+# Copy in the dll files to be ready to run
+FROM build-env as publish
+WORKDIR /app
+COPY --from=build-env /app/artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug /app/artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug
+COPY --from=build-env /app/sdk/eventhub/Azure.Messaging.EventHubs.Shared /app/sdk/evethub/Azure.Messaging.EventHubs.Shared
+COPY --from=build-env /app/sdk/eventhub/Azure.Messaging.EventHubs/stress /app/sdk/eventhub/Azure.Messaging.EventHubs/stress
```

Comment:
I'm pretty sure this is a "me" problem because I don't have any of the other NET repo build configs set up.

```
--> 9ab93d936cd
[1/2] STEP 14/14: COPY /artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug /app/bin/Debug
Error: error building at STEP "COPY /artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug /app/bin/Debug": checking on sources under "/home/ben/sdk/azure-sdk-for-net": copier: stat: "/artifacts/bin/Azure.Messaging.EventHubs.Stress/Debug": no such file or directory
```

Reply:
Also it may be helpful to add an `ENTRYPOINT` or `RUN` directive to support local `docker run` commands easily (unless I'm reading the dockerfile wrong).

Reply:
@m-redding thinking more about my first comment, is it expected that these dockerfile commands will work in a fresh/from scratch environment? Eventually we will want the stress test to be able to be deployable in CI to the "stress-prod" environment, meaning the docker build command alone can get everything built the way it needs to. Currently any stress tests checked in to main will get auto-deployed weekly to that env.

Reply:
I had hoped that they would, so I'll try to figure out what is causing that error, because yes, we definitely will want them to be deployable in CI eventually.

Reply:
> Also it may be helpful to add an `ENTRYPOINT` or `RUN` directive to support local `docker run` commands easily (unless I'm reading the dockerfile wrong).

@benbp Just to clarify since I'm still a little new to docker, what would be the usual use case for running `docker run` ? Running all the tests in succession?

Reply:
@m-redding One example use case could be: let's say your program was working locally, but wasn't working in the cluster, possibly because of a missing file or dependency. It would be much easier to iterate on a bug like this if you were able to do a dev/test loop like `docker build -t testing && docker run -it testing` vs. having to `push the image/re-deploy the stress test/wait for image download/wait for resource deployment/wait for pod start/run kubectl logs`.

So if you set `ENTRYPOINT` and `CMD` like:

```
ENTRYPOINT ["Azure.Messaging.EventHubs.Stress.exe", "--tests"]
CMD ["EventProd"]
```

Then `docker run -it <image tag>` would execute `Azure.Messaging.Eventhubs.Stress.exe --tests EventProd` in the container on startup. You could then override the scenario like `docker run -it <image tag> BuffProd` which would replace the `CMD` directive and execute `Azure.Messaging.Eventhubs.Stress.exe --tests BuffProd`.

When setting `command` and `args` in the pod yaml spec it tells kubernetes to override anything set in the dockerfile, hence why it's only useful for local dev.